requirements/backend/src/dm/dm.gateway.ts

```diff
@@ -1,26 +1,54 @@
-import { UsePipes, ValidationPipe } from '@nestjs/common';
-import { WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
-import { Server } from 'socket.io';
+import { UseFilters, UsePipes } from '@nestjs/common';
+import {
+  MessageBody,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+  WsException,
+} from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
 import { UserEntity } from 'src/database/entity/entity.user';
-import { UserDataType } from 'src/users/dto/user.data.type.dto';
+import { DmChatDto } from './dm.chat.dto';
+import { WsExceptionFilter } from '../ws.exception.filter';
 import { DmService } from './dm.service';
+import { WsValidationPipe } from '../ws.validation.pipe';
 
 @WebSocketGateway({
   cors: {
     origin: 'http://localhost:4242',
   },
 })
-@UsePipes(new ValidationPipe())
+@UseFilters(new WsExceptionFilter())
+@UsePipes(new WsValidationPipe())
 export class DmGateway {
   constructor(private readonly dmService: DmService) {}
 
   @WebSocketServer()
   server: Server;
 
+  @SubscribeMessage('test')
+  handleTest(@MessageBody() data): void {
+    throw new WsException(data.one);
+    // throw new HttpException(data.one, 400);
+  }
+
+  @SubscribeMessage('dm/msg')
+  async msg(client: Socket, payload: DmChatDto) {
+    await client.join(`dm${payload.targetUid}`);
+    this.server.to(`dm${payload.targetUid}`).emit('dm/msg', payload.msg);
+    await client.leave(`dm${payload.targetUid}`);
```

Comment:
```suggestion
    this.server.to(`dm${client.data.uid}`).emit('dm/msg', payload.msg);
    this.server.to(`dm${payload.targetUid}`).emit('dm/msg', payload.msg);
```
위와 같이 하지 않은 것은 이벤트 발생을 최대한 동시에 시켜주기 위함으로 이해하면 될까요??

Reply:
네.
서로 보는 메시지를 동기화하기 위해서,
프론트 쪽에서도 메시지 보내자마자 바로 띄우는게 아니라
백엔드에서 보내는 걸 띄워주기로 했습니다.